spec/integration/migration_service_spec.rb

```diff
@@ -314,6 +314,50 @@ def setup_validation_time_expectations
     end
   end
 
+  context "compressed archives" do
+    let(:old_person) {
+      FactoryBot.create(:person,
+                        profile:               FactoryBot.build(:profile),
+                        serialized_public_key: archive_private_key.public_key.export,
+                        diaspora_handle:       archive_author)
+    }
+
+    it "uncompresses gz archive" do
+      gz_compressed_file = create_gz_archive
+      service = MigrationService.new(gz_compressed_file, new_username)
+      expect(service.only_import?).to be_truthy
+    end
+
+    it "uncompresses zip archive" do
+      zip_ccompressed_file = create_zip_archive
+      service = MigrationService.new(zip_ccompressed_file, new_username)
+      expect(service.only_import?).to be_truthy
+    end
+
+    def create_gz_archive
+      target_path = File.dirname(archive_file.path)
+      target_file = File.join(target_path, "archive.json.gz")
```

Comment:
You should ensure that compressed files also are removed again after the test, so best is probably to also use a `Tempfile` instead.